### DISCUSSION TITLE
How can I evaluate averages in parts of the domain not at a boundary?

### DISCUSSION THREAD

**Discussion by ChristianBauerEng** [su2code/SU2#1877] [Q&A] [answered]
Hi all,

I'm running 2D axisymmetric cases in which I'm interested in the average fluid temperatures in parts of the domain. For example here i the part marked red.
![grafik](https://user-images.githubusercontent.com/42602495/210860583-4e521476-096e-454c-9824-33de652b8e40.png)
Of course this is easy to do in post-processing, but I'd like to use these mean values as an optimization goal and for convergence monitoring, so I need them at runtime.

In previous simulations I used Ansys Fluent on CGNS meshes. In these CGNS meshes the domain was internally divided and each of the domain parts could be given a dedicated name. The domain neighbours were connected by conformal "internal" boundary conditions. These domain markers and BCs could be used for evaluating properties at run-time.

The mesh shown above is actually generated in the same way, but prior to exporting in native SU2 format the various domains are joined and the interface between them lost.

So, TL;DR
Is there a way to calculate cell averages in parts of the domain at runtime?

Regards,
Christian

**Comment by bigfooted** [comment, marked as answer]
su2 supports internal markers, called MARKER_INTERNAL, see also https://github.com/su2code/SU2/issues/986

**Reply by ChristianBauerEng** [reply]
Thank you, that was exactly what I was looking for!
I'll have to teach Pointwise how to do that (probably by manually adding the marker in a post-export step), but I'll give it a try.

Since the documentation regarding mesh seems to be outdated ("For a 2D mesh, only line elements are supported along the boundaries.") I'd like to ask if it's also possible to assign markers to 2D elements in a 2D mesh. That would allow me to calculate volume-averages over a subset of the domain, not just averages over a cross section.

**Reply by pcarruscag** [reply]
If you want a volume average, we do not have them. Using marker internal will give you a surface average.

**Reply by ChristianBauerEng** [reply]
Thank you, I can live with that for now.

If it makes sense I'll open a feature request issue to further discuss if volume-averages may be worthwile to implement in the future.

**Reply by ChristianBauerEng** [reply]
For anyone working with Pointwise and facing the problem that you cannot create an internal boundary:
Create two seperate domains with a conformal interface where you want the "internal" boundary to be. Select SU2 as the solver (CAE->Select Solver). Even though Pointwise won't allow you to graphically select the internal boundary for BC creation you can do so by scripting:
```
 set BC [pw::BoundaryCondition create]
  $BC setName $Name
  $BC apply [list [list $Domain $Connector]]
```
When exporting the mesh, Pointwise will merge the domains but keep the boundary markers. At least in 2D this did the trick for me!

**Reply by ChristianBauerEng** [reply]
Small correction:
Unfortunately setting a BC as mentioned above indeed causes Pointwise to duplicate the nodes on this marker and create two seperated mesh zones. Additionally, another "Unspecified" marker occurs in the mesh.

I could rectify this by manually post-processing the resulting mesh, following these steps:

- Read mesh file
-  Iterate over all nodes referenced in marker "Unspecified"
- Find closest point next to the current one. Distance should in fact be "0"
- Replace all occurences of the Node-ID found in "Unspecified" with the Node-ID of the duplicate point
- Delete Marker "Unspecified" and all nodes referenced by it
- Find and delete all now vacant nodes, that are no longer referenced by any marker or element
- Update Node-IDs in all elements and markers
- Export mesh again

In the end it would probably have been easier to write a custom plugin for Pointwise to write the mesh file, but hey, now it works.
